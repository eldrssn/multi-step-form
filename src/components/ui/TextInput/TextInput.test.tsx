import { fireEvent, render, waitFor } from '@testing-library/react';
import { TextInput } from './';
import { Formik } from 'formik';

test('Input', async () => {
  const { getByPlaceholderText, container } = render(
    <Formik initialValues={{ name: '' }} onSubmit={() => {}}>
      <TextInput
        input={{
          type: 'text',
          name: 'name',
          label: 'Name',
          placeholder: 'Enter your name',
        }}
      />
    </Formik>
  );

  const input = getByPlaceholderText(/Enter your name/i);
  fireEvent.change(input, { target: { value: 'Jane' } });
  fireEvent.blur(input);

  await waitFor(() => {
    expect(input).toHaveValue('Jane');
    expect(container).toMatchSnapshot();
  });
});
