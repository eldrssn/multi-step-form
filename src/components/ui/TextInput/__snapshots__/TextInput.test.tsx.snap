// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 1`] = `
<div>
  <div
    class="input"
  >
    <div
      class="label_box"
    >
      <label
        for="name"
      >
        Name
      </label>
    </div>
    <input
      id="name"
      label="Name"
      name="name"
      placeholder="Enter your name"
      type="text"
      value="Jane"
    />
  </div>
</div>
`;
